.gitmodules

```diff
@@ -1,6 +1,7 @@
 [submodule "src/libmongoc"]
 	path = src/libmongoc
-	url = https://github.com/mongodb/mongo-c-driver.git
+	url = https://github.com/jmikola/mongo-c-driver.git
+	branch = cdriver-3909
```

Comment:
Adding `branch` here shouldn't be necessary.

[Updating the submodule](https://github.com/mongodb/mongo-php-driver/blob/master/CONTRIBUTING.md#update-libmongoc-submodule) should only entail entering the `src/libmongoc` directory, checking out the appropriate commit (in this case, the commit on my fork), and then going back to the root directory of the mongo-php-driver repository and adding `src/libmongoc` to your commit.

Your PR is missing a change to `src/libmongoc`, so I assume you missed the last step.

For example, if you look at https://github.com/mongodb/mongo-php-driver/commit/ac4935338a65d7f3370befd5c245e1be68f9eeba you'll see a change for "Submodule libmongoc updated 155 files". That is the result of me checking out a different commit inside of `src/libmongoc` and then going back to the root PHPC repository and adding that path to a commit. Before doing so `git status` should also indicate that `src/libmongoc` has changes, which indicates that you're able to add it to a commit.

Reply:
I also want to confirm that you were able to successfully add my libmongoc fork as a remote (e.g. `git remote add...`) within `src/libmongoc`, fetch from it (to pull in references to its branches/commits), and checkout the `cdriver-3909` branch on your local machine.

If so, you should be left with `src/libmongoc` pointing to https://github.com/jmikola/mongo-c-driver/commit/be7016760646e0ca129956da1933e3f5983db23b. The branch name itself is just a pointer to that commit, and the commit hash is all that is used by git when updating the submodule.

I don't recall ever using a `branch` option in `.gitmodules`. Changing the `url` should be all that is necessary as it will ensure git refers to the correct repository (my fork in this case) that contains the commit hash you're updating `src/libmongoc` to reference.

Reply:
Oh got it! Yes, I added your libmongoc fork as a remote; the part I was missed was adding the `src/libmongoc` to my commit.